test/Dapr.E2E.Test/DaprApp.cs

```diff
@@ -0,0 +1,99 @@
+﻿// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Dapr.E2E.Test
+{
+    public class DaprApp
+    {
+        // private string pathSeparator;
+        static string shellExeName = SetShell();
+        private string appId;
+        private int appPort;
+        const string outputToMatchOnStart = "You're up and running";
+        const string outputToMatchOnStop = "app stopped successfully";
+
+        static char pathSeparator = System.IO.Path.DirectorySeparatorChar;
+
+        private static string SetShell()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return "cmd.exe";
+            }
+            return "/bin/bash";
+        }
+
+        public DaprApp(string appId, int appPort = 0)
+        {
+            this.appId = appId;
+            this.appPort = appPort;
+        }
+
+        public (string, string) Start()
+        {
+            var (httpPort, grpcPort, metricsPort) = GetFreePorts();
+            var componentsPath = string.Format($"..{DaprApp.pathSeparator}..{DaprApp.pathSeparator}..{DaprApp.pathSeparator}..{DaprApp.pathSeparator}..{DaprApp.pathSeparator}test{DaprApp.pathSeparator}Dapr.E2E.Test{DaprApp.pathSeparator}components");
+            var daprStartCommand = string.Format($"dapr run --app-id {appId} --dapr-http-port {httpPort} --dapr-grpc-port {grpcPort} --metrics-port {metricsPort} --components-path .{DaprApp.pathSeparator}{componentsPath}");
```

Comment:
`.{DaprApp.pathSeparator}{componentsPath}` is a little confusing, why not just resolve to a full path? https://docs.microsoft.com/en-us/dotnet/api/system.io.path?view=net-5.0 would help